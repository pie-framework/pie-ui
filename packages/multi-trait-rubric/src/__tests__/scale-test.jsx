import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Scale from '../scale';

describe('Scale', () => {
  let wrapper, scale;

  const mkWrapper = (scaleOptions = {}) => {
    scale = {
      excludeZero: false,
      maxPoints: 1,
      scorePointsLabels: ['A', 'B'],
      traitLabel: 'Trait',
      traits: [
        {
          name: 'Trait 1',
          description: 'Trait 1 Description',
          scorePointsDescriptors: ['Descriptor 0', 'Descriptor 1'],
          standards: [],
        }
      ],
      ...scaleOptions
    };

    return mount(<Scale
      scale={scale}
      scaleIndex={1}
      classes={{}}
    />);
  };

  beforeEach(() => {
    wrapper = mkWrapper();
  });

  describe('snapshots', () => {
    it('renders without 0', () => {
      const wrapper = mkWrapper({ excludeZero: true });
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders with 0', () => {
      const wrapper = mkWrapper();
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders with no score points', () => {
      const wrapper = mkWrapper({ excludeZero: true, maxPoints: 0 });
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders with no traits', () => {
      const wrapper = mkWrapper({ traits: [] });
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
