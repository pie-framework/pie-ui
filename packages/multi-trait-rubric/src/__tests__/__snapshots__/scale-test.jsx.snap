// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scale snapshots renders with 0 1`] = `
<WithStyles(Scale)
  classes={Object {}}
  scale={
    Object {
      "excludeZero": false,
      "maxPoints": 1,
      "scorePointsLabels": Array [
        "A",
        "B",
      ],
      "traitLabel": "Trait",
      "traits": Array [
        Object {
          "description": "Trait 1 Description",
          "name": "Trait 1",
          "scorePointsDescriptors": Array [
            "Descriptor 0",
            "Descriptor 1",
          ],
          "standards": Array [],
        },
      ],
    }
  }
  scaleIndex={1}
>
  <Scale
    classes={
      Object {
        "pointLabel": "Scale-pointLabel-3",
        "scorePointHeader": "Scale-scorePointHeader-2",
        "scorePointValue": "Scale-scorePointValue-4",
        "table": "Scale-table-1",
      }
    }
    scale={
      Object {
        "excludeZero": false,
        "maxPoints": 1,
        "scorePointsLabels": Array [
          "A",
          "B",
        ],
        "traitLabel": "Trait",
        "traits": Array [
          Object {
            "description": "Trait 1 Description",
            "name": "Trait 1",
            "scorePointsDescriptors": Array [
              "Descriptor 0",
              "Descriptor 1",
            ],
            "standards": Array [],
          },
        ],
      }
    }
    scaleIndex={1}
  >
    <table
      className="Scale-table-1"
      key="scale-1"
    >
      <thead>
        <tr>
          <th>
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Trait",
                }
              }
            />
          </th>
          <th
            key="table-header-0"
          >
            <table
              className="Scale-scorePointHeader-2"
            >
              <thead>
                <tr>
                  <td
                    className="Scale-scorePointValue-4"
                  >
                    1 point
                  </td>
                </tr>
              </thead>
            </table>
          </th>
          <th
            key="table-header-1"
          >
            <table
              className="Scale-scorePointHeader-2"
            >
              <thead>
                <tr>
                  <td
                    className="Scale-scorePointValue-4"
                  >
                    0 points
                  </td>
                </tr>
              </thead>
            </table>
          </th>
        </tr>
      </thead>
      <tbody>
        <WithStyles(Trait)
          excludeZero={false}
          key="trait_1_0"
          scaleIndex={1}
          scorePointsValues={
            Array [
              1,
              0,
            ]
          }
          showDescription={false}
          showStandards={false}
          trait={
            Object {
              "description": "Trait 1 Description",
              "name": "Trait 1",
              "scorePointsDescriptors": Array [
                "Descriptor 0",
                "Descriptor 1",
              ],
              "standards": Array [],
            }
          }
          traitIndex={0}
        >
          <Trait
            classes={
              Object {
                "noDescription": "Trait-noDescription-6",
                "trait": "Trait-trait-5",
                "traitName": "Trait-traitName-7",
              }
            }
            excludeZero={false}
            scaleIndex={1}
            scorePointsValues={
              Array [
                1,
                0,
              ]
            }
            showDescription={false}
            showStandards={false}
            trait={
              Object {
                "description": "Trait 1 Description",
                "name": "Trait 1",
                "scorePointsDescriptors": Array [
                  "Descriptor 0",
                  "Descriptor 1",
                ],
                "standards": Array [],
              }
            }
            traitIndex={0}
          >
            <tr
              key="scale-1-trait-0"
            >
              <td
                className="Trait-trait-5 Trait-traitName-7"
              >
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Trait 1",
                    }
                  }
                />
              </td>
              <td
                key="table-cell-0"
              >
                <div
                  className=""
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Descriptor 1",
                    }
                  }
                />
              </td>
              <td
                key="table-cell-1"
              >
                <div
                  className=""
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Descriptor 0",
                    }
                  }
                />
              </td>
            </tr>
          </Trait>
        </WithStyles(Trait)>
      </tbody>
    </table>
  </Scale>
</WithStyles(Scale)>
`;

exports[`Scale snapshots renders with no score points 1`] = `
<WithStyles(Scale)
  classes={Object {}}
  scale={
    Object {
      "excludeZero": true,
      "maxPoints": 0,
      "scorePointsLabels": Array [
        "A",
        "B",
      ],
      "traitLabel": "Trait",
      "traits": Array [
        Object {
          "description": "Trait 1 Description",
          "name": "Trait 1",
          "scorePointsDescriptors": Array [
            "Descriptor 0",
            "Descriptor 1",
          ],
          "standards": Array [],
        },
      ],
    }
  }
  scaleIndex={1}
>
  <Scale
    classes={
      Object {
        "pointLabel": "Scale-pointLabel-3",
        "scorePointHeader": "Scale-scorePointHeader-2",
        "scorePointValue": "Scale-scorePointValue-4",
        "table": "Scale-table-1",
      }
    }
    scale={
      Object {
        "excludeZero": true,
        "maxPoints": 0,
        "scorePointsLabels": Array [
          "A",
          "B",
        ],
        "traitLabel": "Trait",
        "traits": Array [
          Object {
            "description": "Trait 1 Description",
            "name": "Trait 1",
            "scorePointsDescriptors": Array [
              "Descriptor 0",
              "Descriptor 1",
            ],
            "standards": Array [],
          },
        ],
      }
    }
    scaleIndex={1}
  >
    <table
      className="Scale-table-1"
      key="scale-1"
    >
      <thead>
        <tr>
          <th>
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Trait",
                }
              }
            />
          </th>
        </tr>
      </thead>
      <tbody>
        <WithStyles(Trait)
          excludeZero={true}
          key="trait_1_0"
          scaleIndex={1}
          scorePointsValues={Array []}
          showDescription={false}
          showStandards={false}
          trait={
            Object {
              "description": "Trait 1 Description",
              "name": "Trait 1",
              "scorePointsDescriptors": Array [
                "Descriptor 0",
                "Descriptor 1",
              ],
              "standards": Array [],
            }
          }
          traitIndex={0}
        >
          <Trait
            classes={
              Object {
                "noDescription": "Trait-noDescription-6",
                "trait": "Trait-trait-5",
                "traitName": "Trait-traitName-7",
              }
            }
            excludeZero={true}
            scaleIndex={1}
            scorePointsValues={Array []}
            showDescription={false}
            showStandards={false}
            trait={
              Object {
                "description": "Trait 1 Description",
                "name": "Trait 1",
                "scorePointsDescriptors": Array [
                  "Descriptor 0",
                  "Descriptor 1",
                ],
                "standards": Array [],
              }
            }
            traitIndex={0}
          >
            <tr
              key="scale-1-trait-0"
            >
              <td
                className="Trait-trait-5 Trait-traitName-7"
              >
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Trait 1",
                    }
                  }
                />
              </td>
            </tr>
          </Trait>
        </WithStyles(Trait)>
      </tbody>
    </table>
  </Scale>
</WithStyles(Scale)>
`;

exports[`Scale snapshots renders with no traits 1`] = `
<WithStyles(Scale)
  classes={Object {}}
  scale={
    Object {
      "excludeZero": false,
      "maxPoints": 1,
      "scorePointsLabels": Array [
        "A",
        "B",
      ],
      "traitLabel": "Trait",
      "traits": Array [],
    }
  }
  scaleIndex={1}
>
  <Scale
    classes={
      Object {
        "pointLabel": "Scale-pointLabel-3",
        "scorePointHeader": "Scale-scorePointHeader-2",
        "scorePointValue": "Scale-scorePointValue-4",
        "table": "Scale-table-1",
      }
    }
    scale={
      Object {
        "excludeZero": false,
        "maxPoints": 1,
        "scorePointsLabels": Array [
          "A",
          "B",
        ],
        "traitLabel": "Trait",
        "traits": Array [],
      }
    }
    scaleIndex={1}
  >
    <table
      className="Scale-table-1"
      key="scale-1"
    >
      <thead>
        <tr>
          <th>
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Trait",
                }
              }
            />
          </th>
          <th
            key="table-header-0"
          >
            <table
              className="Scale-scorePointHeader-2"
            >
              <thead>
                <tr>
                  <td
                    className="Scale-scorePointValue-4"
                  >
                    1 point
                  </td>
                </tr>
              </thead>
            </table>
          </th>
          <th
            key="table-header-1"
          >
            <table
              className="Scale-scorePointHeader-2"
            >
              <thead>
                <tr>
                  <td
                    className="Scale-scorePointValue-4"
                  >
                    0 points
                  </td>
                </tr>
              </thead>
            </table>
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </Scale>
</WithStyles(Scale)>
`;

exports[`Scale snapshots renders without 0 1`] = `
<WithStyles(Scale)
  classes={Object {}}
  scale={
    Object {
      "excludeZero": true,
      "maxPoints": 1,
      "scorePointsLabels": Array [
        "A",
        "B",
      ],
      "traitLabel": "Trait",
      "traits": Array [
        Object {
          "description": "Trait 1 Description",
          "name": "Trait 1",
          "scorePointsDescriptors": Array [
            "Descriptor 0",
            "Descriptor 1",
          ],
          "standards": Array [],
        },
      ],
    }
  }
  scaleIndex={1}
>
  <Scale
    classes={
      Object {
        "pointLabel": "Scale-pointLabel-3",
        "scorePointHeader": "Scale-scorePointHeader-2",
        "scorePointValue": "Scale-scorePointValue-4",
        "table": "Scale-table-1",
      }
    }
    scale={
      Object {
        "excludeZero": true,
        "maxPoints": 1,
        "scorePointsLabels": Array [
          "A",
          "B",
        ],
        "traitLabel": "Trait",
        "traits": Array [
          Object {
            "description": "Trait 1 Description",
            "name": "Trait 1",
            "scorePointsDescriptors": Array [
              "Descriptor 0",
              "Descriptor 1",
            ],
            "standards": Array [],
          },
        ],
      }
    }
    scaleIndex={1}
  >
    <table
      className="Scale-table-1"
      key="scale-1"
    >
      <thead>
        <tr>
          <th>
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Trait",
                }
              }
            />
          </th>
          <th
            key="table-header-0"
          >
            <table
              className="Scale-scorePointHeader-2"
            >
              <thead>
                <tr>
                  <td
                    className="Scale-scorePointValue-4"
                  >
                    1 point
                  </td>
                </tr>
              </thead>
            </table>
          </th>
        </tr>
      </thead>
      <tbody>
        <WithStyles(Trait)
          excludeZero={true}
          key="trait_1_0"
          scaleIndex={1}
          scorePointsValues={
            Array [
              1,
            ]
          }
          showDescription={false}
          showStandards={false}
          trait={
            Object {
              "description": "Trait 1 Description",
              "name": "Trait 1",
              "scorePointsDescriptors": Array [
                "Descriptor 0",
                "Descriptor 1",
              ],
              "standards": Array [],
            }
          }
          traitIndex={0}
        >
          <Trait
            classes={
              Object {
                "noDescription": "Trait-noDescription-6",
                "trait": "Trait-trait-5",
                "traitName": "Trait-traitName-7",
              }
            }
            excludeZero={true}
            scaleIndex={1}
            scorePointsValues={
              Array [
                1,
              ]
            }
            showDescription={false}
            showStandards={false}
            trait={
              Object {
                "description": "Trait 1 Description",
                "name": "Trait 1",
                "scorePointsDescriptors": Array [
                  "Descriptor 0",
                  "Descriptor 1",
                ],
                "standards": Array [],
              }
            }
            traitIndex={0}
          >
            <tr
              key="scale-1-trait-0"
            >
              <td
                className="Trait-trait-5 Trait-traitName-7"
              >
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Trait 1",
                    }
                  }
                />
              </td>
              <td
                key="table-cell-0"
              >
                <div
                  className=""
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Descriptor 0",
                    }
                  }
                />
              </td>
            </tr>
          </Trait>
        </WithStyles(Trait)>
      </tbody>
    </table>
  </Scale>
</WithStyles(Scale)>
`;
